### DISCUSSION TITLE
Relocating packages from /opt/homebrew to /usr/local/bin

### DISCUSSION THREAD

**Discussion by pjpraw** [Homebrew/discussions#3336] [Getting started]
I'm on Mac M1, MacOS Big Sur version 11.4.
I successfully installed homebrew and ran few commands: `brew install gcc`, `brew install cbc` and `brew install unzip`.
My objective is to run the OSeMOSYS model via clicSANDMAC.

I understand that, for the Mac M1, packages are installed in /opt/homebrew by default. However, I don't know where I can search and open these packages. I want to relocate the packages to /usr/local/bin.

I read many posts and they advised changing the shell environment which I did not fully understand **(I'm a beginner here)**. For example, some recommended
- `echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> /Users/<my username>/.zprofile`
   Then, `eval "$(/opt/homebrew/bin/brew shellenv)"`
   I believe that these two commands should be run after installing homebrew (as suggested in other users' terminal). Unexpectedly, this suggestion did not show up on my terminal
- `export PATH="/usr/local/bin:/usr/local/sbin:~/bin:$PATH"`
- `sudo`

How can I identify which method I should use? And, do I need to reinstall homebrew and/or packages before following a proposed method?

Thank you for your guidance

**Comment by SMillerDev** [comment]
>  I believe that these two commands should be run after installing homebrew (as suggested in other users' terminal). Unexpectedly, this suggestion did not show up on my terminal

If you installed Homebrew through the script on `brew.sh` they definitely showed at the end of the script output.

`export PATH="/usr/local/bin:/usr/local/sbin:~/bin:$PATH"` or `sudo` won't do anything.

>  I don't know where I can search and open these packages. 

In `/opt/homebrew`? Unless you mean running executables from them, in which case you should run the `eval` commands you mentioned.

> I want to relocate the packages to /usr/local/bin.

This is a very bad idea and will break Homebrew completely. Any problems you are encountering have to be fixed by pointing the programs you use to the right location. Moving Homebrew will just cause a lot of other problems.

>  My objective is to run the OSeMOSYS model via clicSANDMAC.

I think the best course of action is to ask the developers of those tools for help.

**Reply by pjpraw** [reply]
Dear @SMillerDev 

Thank you for your clarification. However, I still cannot provide the path to /usr/local/bin

I uninstalled homebrew and installed it again. The following .txt file provides a detail shown in my terminal. Did I make any mistake which results in an incomplete end of the script output?
[no_echo_eval.txt](https://github.com/Homebrew/discussions/files/8853683/no_echo_eval.txt)

Thank you for your guidance

**Reply by carlocab** [reply]
It wasn't shown because `/opt/homebrew/bin` is already in your `PATH`, so the installer assumes that you don't need to be told to add it anymore.

**Reply by pjpraw** [reply]
Dear @carlocab

Since running the OSeYMOSYS model in the clicSANDMac require the gcc, cbc, unzip command to be located in the `/usr/local/bin`.
Can you please advise me on how I can change the path from `/opt/homebrew/bin` to `/usr/local/bin`?

Thank you :)

**Reply by carlocab** [reply]
@pjpraw I suggest that you heed the advice you've been given above:

> > I want to relocate the packages to /usr/local/bin.
> 
> This is a very bad idea and will break Homebrew completely. Any problems you are encountering have to be fixed by pointing the programs you use to the right location. Moving Homebrew will just cause a lot of other problems.
> 
> > My objective is to run the OSeMOSYS model via clicSANDMAC.
> 
> I think the best course of action is to ask the developers of those tools for help.

If the software you're trying to run expects things to only be in `/usr/local/bin`, then that seems like a bug in the aforementioned software. `gcc`, `cbc`, and `unzip` can be installed in arbitrary locations, so it is wrong to only expect them in a single location.

**Reply by SMillerDev** [reply]
> Since running the OSeYMOSYS model in the clicSANDMac require the gcc, cbc, unzip command to be located in the /usr/local/bin.

That should be fixed in those tools then, moving Homebrew will break Homebrew so that's not really a feasible solution.

**Reply by pjpraw** [reply]
Dear @carlocab and @SMillerDev 

I have been figuring out this issue for a few days. I appreciate your prompt clarification.
I will contact the tool developers and continue to seek their advice.
